packages/typespec-azure-resource-manager/src/operations.ts

```diff
@@ -158,6 +180,70 @@ function setResourceLifecycleOperation(
   };
 
   operations.lifecycle[kind] = operation as ArmResourceOperation;
+  setArmOperationIdentifier(context.program, target, resourceType, {
+    name: target.name,
+    kind: kind,
+    operation: target,
+    operationGroup: target.interface.name,
+  });
+}
+
+export function getArmResourceOperationList(
+  program: Program,
+  resourceType: Model,
+): Set<ArmOperationIdentifier> {
+  if (!program.stateMap(ArmStateKeys.resourceOperationList).has(resourceType)) {
```

Comment:
You can use useStateMap and useStateSet to define setter and getter to the state and keep a typed version to make sure the setter and getter don't see the data differently. 

You can check http lib for a lot of examples or I migrated some of azure core Lro status

Reply:
Yes, but in this case, these operations are only used by our internal decorators.  Happy to change if we want to keep an abstraction over the actual keys for future-proofing.

Reply:
yeah just feels like its safer, so many decorators historically set different data that the getter would expect(specially around undefined). This is also IMO just cleaner and also shorter(if you do actually want an exposed getter/setter which is not your case)